### ISSUE TITLE
BGW320-500 downgrade not possible?

### ISSUE THREAD

**Issue by hollanbm** [0x888e/certs#3]
My BGW320-500 is on firmware v6.28.7, and when I try to downgrade to v3.18.1, I get an error that says “Firmware upgrade failed due to version incompatibility”

I even rebooted and immediately uploaded the firmware post startup, like the readme calls out, but still no luck.

Am I missing something obvious? Or am I stuck, unable to downgrade?

thanks!

**Comment by BlakerBlaker** [comment]
I have not been able to find any downloads for the older firmware. Can anyone share a link? I have the 320-505 and my current FW is 6.28.7 so hopefully won't experience the same as hollanbm but can at least test it to verify if this is an issue with new FW or isolate to your hardware

**Comment by miles267** [comment]
> My BGW320-500 is on firmware v6.28.7, and when I try to downgrade to v3.18.1, I get an error that says “Firmware upgrade failed due to version incompatibility”
> 
> I even rebooted and immediately uploaded the firmware post startup, like the readme calls out, but still no luck.
> 
> Am I missing something obvious? Or am I stuck, unable to downgrade?
> 
> thanks!

Did you find a solution for this?  Were you able to downgrade fw from 6.28.7?

**Comment by bholland-bh** [comment]
> > My BGW320-500 is on firmware v6.28.7, and when I try to downgrade to v3.18.1, I get an error that says “Firmware upgrade failed due to version incompatibility”
> 
> > 
> 
> > I even rebooted and immediately uploaded the firmware post startup, like the readme calls out, but still no luck.
> 
> > 
> 
> > Am I missing something obvious? Or am I stuck, unable to downgrade?
> 
> > 
> 
> > thanks!
> 
> 
> 
> Did you find a solution for this?  Were you able to downgrade fw from 6.28.7?

Nope, no solution. AFAICT it's not possible to downgrade any longer

**Comment by miles267** [comment]
Can you please try to downgrade from 6.28.7 to 3.17? Instead of 3.18?

**Comment by hollanbm** [comment]
> Can you please try to downgrade from 6.28.7 to 3.17? Instead of 3.18?

I’ve been upgrade to 6.30.5 since, and 3.18.1 is the only version I could find. Do you have a link to 3.17?

**Comment by miles267** [comment]
Please see if it'll let you use this one: https://archive.org/download/ATTfirmware/bgw320-505/spTurquoise320-505_3.17.5_dnvpnP_021_sec.bin.  Let us know if you're able to downgrade to it.

**Comment by bholland-bh** [comment]
> Please see if it'll let you use this one: https://archive.org/download/ATTfirmware/bgw320-505/spTurquoise320-505_3.17.5_dnvpnP_021_sec.bin.  Let us know if you're able to downgrade to it.

I have a BGW320-500, so I can't use that firmware file

**Comment by miles267** [comment]
@bholland-bh apologies.  please try this one? https://archive.org/download/ATTfirmware/bgw320-500/spTurquoise320-500_3.17.5_dnvpnP_021_sec.bin

**Comment by hollanbm** [comment]
> @bholland-bh apologies.  please try this one? https://archive.org/download/ATTfirmware/bgw320-500/spTurquoise320-500_3.17.5_dnvpnP_021_sec.bin

🤦‍♂️ didn't realize I was responding from the wrong account.

Thanks for the link, I will give this a try later tonight once the family has all gone to bed.

**Comment by c-herz** [comment]
Looks like both links are down. Can someone drop a new link? Maybe try google drive or something?

**Comment by Mr1070** [comment]
Ok so the solution for the downgrade is 

Unplug the fiber 
Go to the upgrade page and upgrade to 3.17
(The moment you see the browser is sending the firmware hold down the WPS button the front of the router)
You should see the browser go to the successful upgrading page and start the countdown post install/boot
The next firmware is harder to do because it he router will be in a boot loop

Set up a couple of pages on the upgrade page.  The moment you can see that the router is booted 
Using ```ping -t 192.168.1.254``` it will show TTL of 64 

The pages are live and tou can do the same procedure for 3.18

**Comment by vvarga007** [comment]
@hollanbm Were you able to downgrade from 6x.x.?

**Comment by hollanbm** [comment]
> @hollanbm Were you able to downgrade from 6x.x.?

Yes I was. It was very finicky. But I basically had to do what the README described, rebooting and flashing asap

**Comment by e1342** [comment]
I'm having difficulty back leveling as well; it's most likely [PEBKAC](https://en.wikipedia.org/wiki/User_error#PEBKAC/PEBCAK/PICNIC) and I'm hoping you can point out what I missed. My BGW320-500 is currently running firmware version 6.34.7.

Per the README, I hard coded my local IP and made sure only power and my laptop (WIFI off) were plugged in to the router. I left ping running continuously in the background.

I uploaded the 3.17.5 firmware mentioned in the README, acquired via #7 with MD5 and SHA1 matching the README. After a few minutes, the router rebooted.

As soon as ping responded, I refreshed the page. After logging back in, the screen indicated that it was updating for an estimated NULL minutes, and the page had no place allowing input any kind (for going to 3.18.1). After another minute or so of refreshing the page, I finally got back to the version of the screen that allows uploading a firmware, but that screen showed the current firmware was still the same 6.34.7 I'd had before doing anything in the first place! No reboots occurred, and the continuous ping I left running in the background had not shown any breaks in responses between the upgrading status page and the page allowing to upload a firmware.

In case it was a fluke, I tried uploading 3.17.5 again, but same results.

This is also the first time I've seen mention of pressing the WPS button. What does that do in this context? And just to be clear, you are stating that I should start pressing the WPS button as soon as the firmware starts being uploaded and not waiting to press it as soon as it has finished uploading, right? When should I let go of it? I'm guessing that my having turned off both 2.4ghz and 5ghz wifi radios years ago is not relevant.

Thank you for your time!

**Comment by Mr1070** [comment]
When you are selecting firmware and starting the process .... on successful transfer and flash it goes to a page where it starts to countdown a reboot ... when it doesn't allow you it tells you an error on the same page you selected your firmware.  All of this is of course done offline.  When I watched a "ping 192.168.1.254 -t -w 2"

You can see the moment you can login again before the webpage refreshes and so on. You can also see the moment the transfer ends.  In that moment press the WPS.  The flash has occurred be it is flashed from upload 

You see upload as flash and that is incorrect.  It uploads to memory and then flashes from memory
